cns/restserver/internalapi_test.go

```diff
@@ -0,0 +1,189 @@
+// Copyright 2020 Microsoft. All rights reserved.
+// MIT License
+
+package restserver
+
+import (
+	"fmt"
+	"testing"
+	"strconv"
+
+	"github.com/Azure/azure-container-networking/cns"
+	"github.com/google/uuid"
+)
+
+const (
+	primaryIp = "10.0.0.5"
+	gatewayIp = "10.0.0.1"
+	dockerContainerType = cns.Docker
+)
+
+func TestCreateOrUpdateNetworkContainerInternal(t *testing.T) {
+	// requires more than 30 seconds to run
+	fmt.Println("Test: TestCreateOrUpdateNetworkContainerInternal")
+
+	setEnv(t)
+	setOrchastratorTypeInternal(cns.KubernetesCRD)
+
+	err := createOrUpdateNetworkContainerInternal(t, 2)
+	if err != nil {
+		t.Errorf("Failed to save the goal state for network container of type JobObject "+
+			" due to error: %+v", err)
+		t.Fatal(err)
+	}
+}
+
+func setOrchastratorTypeInternal(orchestratorType string) {
+	fmt.Println("setOrchastratorTypeInternal")
+	svc.state.OrchestratorType = orchestratorType
+}
+
+func createOrUpdateNetworkContainerInternal(t *testing.T, secondaryIpCount int) error {
```

Comment:
does this need to return an error? it seems that at any place where an error would be returned, it instead just fatals.

Reply:
yeah I dont need it, will remove it